src/engine.rs

```diff
@@ -978,7 +980,25 @@ impl Reedline {
     fn buffer_paint(&mut self, prompt: &dyn Prompt) -> Result<()> {
         let (highlighted_line, hint) = self.prepare_buffer_content(prompt);
 
-        self.painter.queue_buffer(highlighted_line, hint)?;
+        let lines = PromptLines::from_strings(&highlighted_line.0, &highlighted_line.1, &hint);
```

Comment:
How about letting `prepare_buffer_content()` directly returning the prompt lines?